packages/cli/src/hooks/analytics/analytics.ts

```diff
@@ -101,61 +58,63 @@ const platformInfoIdentifier = async () => {
   };
 };
 
-const configureAnalyticsClient = (authenticatedClient: AuthenticatedClient) => {
-  const analyticsClient = new CoveoAnalyticsClient({token: analyticsAPIKey});
-  analyticsClient.runtime.storage = storage(authenticatedClient.cfg);
-  return analyticsClient;
-};
+const errorIdentifier = (err?: Error) => ({
+  ...(err && {
+    errorMessage: err.message,
+    errorName: err.name,
+  }),
+});
 
-const storage = (cfg: Config): WebStorage => {
-  return {
-    getItem: async (k) => {
-      const configuration = await cfg.get();
-      return configuration[k] as string;
-    },
-    removeItem: (k) => {
-      cfg.deleteAny(k);
-    },
-    setItem: async (k, v) => {
-      await cfg.setAny(k, v);
-    },
-  };
+const configureAmplitudeClient = () => {
+  // TODO: support proxy
+  const amplitudeClient = init(analyticsAPIKey);
+  return amplitudeClient;
 };
 
 const isLoggedIn = async () => {
   const status = await getAuthenticationStatus();
   return status === AuthenticationStatus.LOGGED_IN;
 };
 
-export type AnalyticsStatus = 'success' | 'failure';
-
-export const buildAnalyticsHook = (
-  cmd: Command,
-  status: AnalyticsStatus,
-  flags: Record<string, unknown>,
+export const buildEvent = (
+  eventName: string,
+  properties: Record<string, unknown>,
   err?: Error
-): AnalyticsHook => {
-  return {
-    commandID: `${status} ${cmd.id}`,
-    config: cmd.config,
-    flags,
-    err,
+): Event => {
+  const analyticsData = {
+    event_type: eventName,
+    event_properties: {
+      ...properties,
+      ...errorIdentifier(err),
+    },
   };
-};
 
-export const buildAnalyticsSuccessHook = (
-  cmd: Command,
-  flags: Record<string, unknown>
-): AnalyticsHook => {
-  return buildAnalyticsHook(cmd, 'success', flags);
+  return analyticsData;
 };
 
-export const buildAnalyticsFailureHook = (
-  cmd: Command,
-  flags: Record<string, unknown>,
-  err: Error | undefined
-): AnalyticsHook => {
-  return buildAnalyticsHook(cmd, 'failure', flags, err);
+export const buildError = (arg: unknown) => {
+  /**
+   * TODO: CDX-660: Make sure to remove any PII from the Error object.
+   *       error.message could contain data that is not allowed to be tracked for non-Trial users
+   *       example: orgID, sourceID, ...
+   */
+  if (arg instanceof Error) {
+    return arg;
+  }
+
+  if (typeof arg === 'string') {
+    return new Error(arg);
+  }
+
+  const schema = {
+    message: 'string',
+    errorCode: 'string',
+    requestID: 'string',
+  };
+  const isErrorFromAPI = validate(arg, schema);
+  return isErrorFromAPI
+    ? new APIError(arg as APIErrorResponse)
+    : new Error('Unknown Error');
 };
```

Comment:
In the end, are we gonna log only the instanceOf `BaseCLIError` (e.g. name, you get the gist)?

If so, I would keep that bit really dead simple with only the arg instanceof Error and call it a day

Reply:
Yes. This will be part of the UA event sanitizing process (CDX-660).
There is no rush since we are only tracking Trial orgs.